### ISSUE TITLE
[Bug] Marko's Magic Football - crash during intro+crash emulation

### ISSUE THREAD

**Issue by nemo93** [snes9xgit/snes9x#231]
Marko's Magic Football (Europe) does crash during intro. This does even crash the emulator! Launch the game and just wait for a few seconds for the intro screen where Marko is running after the ball then emu should stop/crash.

Tested on Snes9x 1.53 + 1.54.1 Windows x64 (as well as Snes9x 1.54.1 libretro+RPi).

**Comment by bonimy** [comment]
This seems to happen during an NMI routine under certain conditions. I haven't pinpointed it yet but am trying to trace the hang. It seems a BRK opcode gets triggered.

Does anyone have any preliminary info on this cause before I dive deeper?

**Comment by SimpleTease** [comment]
Some Canoe hacker named sluffy (see rhdn thread) made a patch that fixed this game. Comment was that inside irq routine:
$80/89AE LDA $4211 DB:7F

Irq never gets cleared correctly and keeps re-firing on cli / rti exit. Hack just added phk-plb to fix problem.

**Comment by bearoso** [comment]
Seems to be fixed with IRQ rework.

**Comment by kps501** [comment]
Another great job! Confirm works as of 2018-05-21 emucr tester build. Played game for some bit.

**Comment by bearoso** [comment]
I've found it's using a spurious plp to pull the IRQ flag off the stack, and this keeps an interrupt from firing, and everything collapses. I'm wondering if it's underflowing or overflowing the stack.

**Comment by kps501** [comment]
We tracked down a possible solution. bsnes log:

** nmi exit **
** 4211 active **
8089e5 plp
8089e6 cli
** irq doesn't immediately fire here **
8089e7 rti
** normal code **
90b8d3

In snes9x 1.51, Op58 = CLI.
We tested:
if( CPU.IRQActive ) CPU.IRQPending = 1

In latest snes9x git, we tested for Op58
	if( CPU.IRQLine )
	{
		CPU.IRQTransition = 0;
		CPU.IRQLine = 0;
		CPU.IRQPending = 1;
	}

These appear to "work". Not at all claiming this is correct behavior, but this is how far we could kick the ball down the pitch. Played several levels without crash.
